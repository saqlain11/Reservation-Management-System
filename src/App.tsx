import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import React,{lazy,Suspense} from "react";
import { Layout, Spin } from "antd";

const AllReservation = React.lazy(() => import("templates/AllReservations"));
const NotFound = React.lazy(() => import("templates/NotFound"));

const App: React.FC = () => {
  return (
    <Router>
        <Suspense
        fallback={
          <Layout
            style={{
              justifyContent: "center",
              alignSelf: "center",
              width: "100vw",
              height: "100vh",
            }}
          >
            <Spin size="large" tip="Loading..."></Spin>
          </Layout>
        }
      >
      <Routes>
        <Route index element={<AllReservation />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
      </Suspense>
    </Router>
  );
};

export default App;
