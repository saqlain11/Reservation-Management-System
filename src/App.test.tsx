import { render } from "@testing-library/react";

import App from "./App";

describe("Main entry point suite", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
