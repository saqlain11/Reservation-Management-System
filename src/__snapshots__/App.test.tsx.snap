// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main entry point suite should render correctly 1`] = `
<DocumentFragment>
  <section
    class="ant-layout css-dev-only-do-not-override-1s3dcof"
    style="justify-content: center; align-self: center; width: 100vw; height: 100vh;"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-1s3dcof"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
      <div
        class="ant-spin-text"
      >
        Loading...
      </div>
    </div>
  </section>
</DocumentFragment>
`;
